src/bootstrap/channel.rs

```diff
@@ -13,7 +14,10 @@ use crate::Build;
 
 pub enum GitInfo {
     /// This is not a git repository.
-    Absent,
+    ///
+    /// If `git-commit-info` can be used for commit information, this is `Some`;
+    /// otherwise, it is `None`.
+    Absent(Option<Info>),
```

Comment:
Hmm, could you explain the naming behind this? This is only used when it's within a tarball, so "for tarball" doesn't seem right. I'll make the change to another variant, though.

Reply:
> This is only used when it's within a tarball, so "for tarball" doesn't seem right.

Can you explain that to me? :sweat: I think it should be "for tarball" exactly because it's in a tarball.

That said, I don't care much about the exact name, as long as it's a different variant that's fine.